nemo/collections/asr/models/enhancement_models.py

```diff
@@ -0,0 +1,440 @@
+            'manifest_filepath': config['manifest_filepath'],
+            'sample_rate': self.sample_rate,
+            'input_key': config['input_key'],
+            'input_channel_selector': config.get('input_channel_selector', None),
+            'target_key': None,
+            'target_channel_selector': None,
+            'batch_size': config['batch_size'],
+            'shuffle': False,
+            'num_workers': config.get('num_workers', min(config['batch_size'], os.cpu_count() - 1)),
+            'pin_memory': True,
+        }
+
+        temporary_dataloader = self._setup_dataloader_from_config(config=DictConfig(dl_config))
+        return temporary_dataloader
+
+    @property
+    def input_types(self) -> Dict[str, NeuralType]:
+        return {
+            "input_signal": NeuralType(
+                ('B', 'T', 'C'), AudioSignal(freq=self.sample_rate)
+            ),  # multi-channel format, channel dimension can be 1 for single-channel audio
+            "input_length": NeuralType(
+                tuple('B'), LengthsType()
+            ),  # multi-channel format, channel dimension can be 1 for single-channel audio
+        }
+
+    @property
+    def output_types(self) -> Dict[str, NeuralType]:
+        return {
+            "output_signal": NeuralType(
+                ('B', 'T', 'C'), AudioSignal(freq=self.sample_rate)
+            ),  # multi-channel format, channel dimension can be 1 for single-channel audio
+        }
+
+    def match_batch_length(self, signal: torch.Tensor, batch_length: int):
+        """Trim or pad the output to match the batch length.
+
+        Args:
+            signal: tensor with shape (B, T, C)
+            batch_length: int
+
+        Returns:
+            Tensor with shape (B, T, C), where T matches the
+            batch length.
+        """
+        signal_length = signal.shape[1]
+        pad_length = batch_length - signal_length
+        pad = (0, 0, 0, pad_length)
+        # pad with zeros or crop
+        return torch.nn.functional.pad(signal, pad, 'constant', 0)
+
+    @typecheck()
+    def forward(self, input_signal, input_length):
+        """
+        Forward pass of the model.
+
+        Args:
+            input_signal: Tensor that represents a batch of raw audio signals,
+                of shape [B, T] or [B, T, C]. T here represents timesteps, with 1 second of audio represented as
+                `self.sample_rate` number of floating point values.
+            input_signal_length: Vector of length B, that contains the individual lengths of the audio
+                sequences.
+
+        Returns:
+        """
+        # TODO: use input_length to mask estimation on padded parts
+        # This is not critical for training, since we can load fixed-length segments for training.
+        # However, it's necessary for some cases, e.g., batched inference.
+        batch_length = input_signal.shape[1]
+
+        # Encoder
+        encoded = self.encoder(input=input_signal)
```

Comment:
here, the encoder is equivalent to something like STFT right? If so, can we change the name to "feature_encoder" or "pre_processor" ?

Reply:
ok. papers have encoder-mask-decoder structure (ASR has different notation where pre-process is feature extractor and encoder picks up bulk of the work). we can stick to paper's convention.